### DISCUSSION TITLE
How to Fire events in usercontrol inside datacontext

### DISCUSSION THREAD

**Discussion by abolzoniSpringa** [AvaloniaUI/Avalonia#7511] [Q&A] [answered]
I'm trying to add a click event for a button inside a usercontrol.

The user control is used as data template for the items in a ListView.
```

	<ListView.ItemTemplate>
		<DataTemplate>
			<Templates:ucProject></Templates:ucProject>
		</DataTemplate>
	</ListView.ItemTemplate>
```

I'm adding a button in user control with a click event
` <Button x:Name="btnOpen"  Click="btnOpen_Click" />`

Then in code behind add the handler
```
 public void btnOpen_Click(object sender, RoutedEventArgs e)
        {
            //code to execute
        }
```

The click event will not fire if the button is clicked.
How can I add events inside the data template?

thanks!

**Comment by ahopper** [comment, marked as answer]
The MVVM approach to this would be to use Command instead of Click and handle it in the view model.

**Reply by abolzoniSpringa** [reply]
thank you, I'm not using MVVM for this control, but adding the command helped me find the error, the click was captured by another control which was supposed to be hidden. 
So the click never actually reached my buttons.

**Comment by workgroupengineering** [comment]
if you don't want to use MVVM see [this](https://docs.avaloniaui.net/docs/input/routed-events#qualified-event-names-in-xaml).
